tests/simulator_test.cpp

```diff
@@ -2,6 +2,7 @@
 #include "main/simulator.h"
 #include "gtest/gtest.h"
 
+// Checks if |TRY_BLOCK| throws right |EXCPETION_TYPE| with defined |Message|
```

Comment:
1. `|MESSAGE|` not `|Message|`
2. This does not say anything about how message is checked. Usually checking for entire message equality is quite brittle in the sense, that when you change the wording in the impl. even by tiny bit, it requires you to change the test. And you do not want that. A better idea is to check for a substring only, and in the tests, only check enough to make sure correct exception is thrown.

Reply:
Changed again and added TODO: for later ;-) note, that this change will be visualised later after push, because I can't review this comment anymore, becuase github marked it as outdated >.>

Reply:
Sounds good.